docs/Coding/internationalization.rst

```diff
@@ -0,0 +1,170 @@
+
+.. code-block:: cpp
+
+    const QString with_number =
+        QString(PL("%1 translation", "%1 translations", count)).arg(count);
+
+In some cases, one needs to disambiguate between different uses of the same string. For instance, ``None`` is
+used in different places to mean "no technology" and "no wonder". This is handled by ``Q_``:
+
+.. code-block:: cpp
+
+    const char *translation = Q_("?wonder:None");
+
+Finally, there is also a macro ``N_()`` that marks a string for translation without doing the translation (it
+returns the original). This is useful in static contexts.
+
+
+Not Translated
+--------------
+
+While most text should be translated, there are a few cases where this is not wanted:
+
+* :strong:`File Formats.` It must be possible to load a saved game independently of the locale that was in
+  use when producing it. The same stands for rulesets, tilesets, and other file formats used by Freeciv21.
+* :strong:`Fatal Error Messages.` If the game encounter such a critical error that it needs to abort, it may
+  not have the resources to produce a translation.
+
+
+Helper Comments
+---------------
+
+When translators work on strings, they are provided with a list taken out of context. They can see the
+original text in English and sometimes the source code --- but most translators cannot read code. In some
+cases, the lack of context makes translation very hard. The string "Close", for instance, can have many
+meanings, with the correct one being inferred from context: "near", "closed", "close (a door)", "stop", ...
+Each meaning calls for a different translation. One can add special comments to the code to help translators
+identify the correct variant:
+
+.. code-block:: cpp
+
+    // TRANS: Close the current window
+    close_button->setText(_("Close"));
+
+The comment will be picked up if it is on the line before the translated text. These comments are typically
+very useful when building text from different parts using placeholders (``%1``, ``%2``, ...). In such cases,
+a comment should be added to explain what the final string looks like:
+
+.. code-block:: cpp
+
+    // TRANS: <Unit> (<Home city>)
+    text += QString::format(_("%1 (%2)")).arg(unit_type_name).arg(home_city_name);
+
+    // TRANS: "HP: 5/20 MP: 5/5" in unit description. Keep short
+    text += QString::format(_("HP: %1/%2 MP: %3/%4"))
+                .arg(hp)
+                .arg(max_hp)
+                .arg(mp)
+                .arg(max_mp);
+
+In complex cases, adding an example or a short explanation also makes the code easier to read.
+
+
+Character Encodings
+-------------------
+
+The way characters are encoded into strings has long been a hot topic of internationalization, and
+language-specific character encodings are still around on some systems. Freeciv21 always uses UTF-8 for data
+files and internal communication (e.g. in the network protocol). ``QString``, the recommended way of storing
+text, uses UTF-16 internally.
```

Comment:
The recommended way of storing text is with the `QString()` constructor function as it uses UTF-16 internally.